### ISSUE TITLE
Reorganize country/format definitions

### ISSUE THREAD

**Issue by istanishev** [Cimpress-MCP/postal-codes-js#5]
* Use "postal-codes-alpha2.json" as default configuration file where everything is defined.
* Fix the numeric3 codes in "postal-codes-alpha2.json" (they should be 3-digits long)
* Generate "postal-codes-alpha3.json" &  "postal-codes-numeric3.json" based on "postal-codes-alpha2.json"
* Retire the json configurations inside mappings folder

**Comment by drzraf** [comment]
I'd advise that a refresh happens from http://download.geonames.org/export/dump/countryInfo.txt
For example "AR" regexp is wrong in this repository but correct in `geonames`.

**Comment by tomhooijenga** [comment]
Is this still wanted? I'd be able to take a shot at this. This package is great, but the mappings are due for some refactoring. There are now three locations to add formats. My last PR showed that this doesn't always go right.

Other things to take in consideration
- Fix the returns: either
  - return strings from an enum
  - return true/false and throw for exceptional situations
- split test data from formats so it isn't loaded in production
- typescript maybe?

Also pinging @ozankaya because they seem to be in charge now

**Comment by ozankaya** [comment]
Yes, this refactoring is still required. Your list above looks good to me. Maybe we can skip TypeScript for now. We have declarations file for TypeScript support.
